### PR TITLE
Refactor: Cleanup config actions

### PR DESCRIPTION
**Pull request by chrisnojima**
This PR generates most config actions

I was going to sagaize this more but didn't want to go down a rabbithole yet
Some stuff requires using put.resolve so needs promises. We could change how this works but it'd need testing which i'm not very interesting in doing right now

### CODE REVIEW

File: shared/actions/config.js [hunk 1]
```diff
@@ -0,0 +1,200 @@
+// @flow
```
Comment (chrisnojima) [inline shared/actions/config.js]: this is actually a move from actions/config/index.js to here

File: shared/actions/config/index.platform.native.js [hunk 2]
```diff
@@ -1,14 +0,0 @@
-// @flow
-import * as Constants from '../../constants/config'
```
Comment (chrisnojima) [inline shared/actions/config/index.platform.native.js]: never used?

File: shared/actions/git/index.js [hunk 3]
```diff
@@ -1,27 +1,24 @@
...
-import * as SettingsConstants from '../../constants/settings'
-
-import type {SagaGenerator} from '../../constants/types/saga'
+import {navigateTo} from '../route-tree'
 
-function* _loadGit(action: Constants.LoadGit): SagaGenerator<any, any> {
-  yield put(Creators.setError(null))
-  yield put(Creators.setLoading(true))
+function* _loadGit(action: Constants.LoadGit): Saga.SagaGenerator<any, any> {
+  yield Saga.put(Creators.setError(null))
```
Comment (chrisnojima) [inline shared/actions/git/index.js]: Using Saga namespace

File: shared/actions/login/saga.js [hunk 4]
```diff
@@ -41,10 +36,11 @@ const InputCancelError = {
 const codePageSelector = ({login: {codePage}}: TypedState) => codePage
 
 function* generateQRCode() {
-  const codePage: AfterSelect<typeof codePageSelector> = yield select(codePageSelector)
```
Comment (chrisnojima) [inline shared/actions/login/saga.js]: lets just get rid of AfterSelect. just use select()

File: shared/reducers/config.js [hunk 5]
```diff
@@ -1,40 +1,9 @@
...
 import * as CommonConstants from '../constants/common'
-import {isMobile} from '../constants/platform'
 
-import type {Action} from '../constants/types/flux'
-
-// Mobile is ready for bootstrap automatically, desktop needs to wait for
-// the installer.
-const readyForBootstrap = isMobile
-
-const initialState: Constants.State = {
```
Comment (chrisnojima) [inline shared/reducers/config.js]: this was moved into constants

**Comment by chrisnojima** [comment]
@keybase/react-hackers this is ready to look at

**Comment by zanderz** [comment]
I don't quite grasp the import, but LGTM - maybe someone else too?

**Comment by chrisnojima** [comment]
@keybase/react-hackers can i get an additional thumb

**Pull request merged**